graphql/data/schema.js

```diff
@@ -1415,6 +1416,66 @@ const createInvitedUsersMutation = mutationWithClientMutationId({
     return invitedUser
   },
 })
+/**
+ * conditionally create squad invite, send automated email
+ */
+const createSquadInvitesMutation = mutationWithClientMutationId({
+  name: 'CreateSquadInvites',
+  inputFields: {
+    inviterId: { type: new GraphQLNonNull(GraphQLString) },
+    invitedEmails: { type: new GraphQLNonNull(GraphQLList(GraphQLString)) },
+    inviterName: { type: new GraphQLNonNull(GraphQLString) },
+    inviterMessage: { type: GraphQLString },
+  },
+  outputFields: {
+    successfulEmailAddresses: {
+      type: new GraphQLList(
+        new GraphQLObjectType({
+          name: 'successfulSquadEmailAddresses',
+          fields: () => ({ email: { type: GraphQLString } }),
+        })
+      ),
+    },
+    existingUserInvited: {
+      type: new GraphQLList(
+        new GraphQLObjectType({
+          name: 'successfulExistingUsers',
+          fields: () => ({ username: { type: GraphQLString } }),
```

Comment:
My concern here is that a client can use this to arbitrarily map email addresses to usernames. It's not a security risk per se, but it's unnecessary data exposure that could cause problems in the future. Will we use these output fields on the client? If so, can we simply return successful/unsuccessful emails (only giving back data that the client gave us)?

Reply:
this makes sense.  We do end up using the  successful existing users usernames in the pending mission table, although we probably get them by refetching the current mission so it's not needed in this response.  I'm currently still limiting the amount of email invites someone can do in 24 hours (can reduce it even more) which should limit how many people someone can ping for.

Reply:
My suggestion is that we only display the (pending) email address until a user joins the mission, at which point we show their username.

Reply:
ok that works!